### DISCUSSION TITLE
Tips (dudas comunes) para realizar la kata 10

### DISCUSSION THREAD

**Discussion by Nukario** [LaunchX-InnovaccionVirtual/CursoIntroPython#220] [Show and tell]
Al momento de realizar este ejercicio puede resultar un poco confuso como realizarlo, así que les dejo como resolví esas confusiones en notebooks de Jupiter.

Cuando vayas a ejecutar instrucciones como esta, únicamente escribe el código sin los ">>>" todo los renglones de abajo es la respuesta que te arroja al ejecutarlo
![image](https://user-images.githubusercontent.com/7990740/153770866-cb3b5e81-dba4-4f9e-b192-acb43fae0c13.png)

Cuando llegues a la parte de definir la función, siempre que te diga que agregues una modificación al código nunca borres el segmento del if (me paso) porque aunque si corre el código no hará nada porque no esta ejecutando la función de "main"

![image](https://user-images.githubusercontent.com/7990740/153770962-c171dc54-4930-43d6-a3fb-47764d9112d6.png)

Tiene que quedar así cada vez que se agrega algo al def main

![image](https://user-images.githubusercontent.com/7990740/153771537-90ad651f-4c28-4ed7-99c1-1dcd76f903f8.png)

En la parte de ejecutar el archivo open.py hay que hacerlo desde la consola, y es muy importante que cambien el directorio a donde tienen el archivo, y que cuando escriban el código no pongan el "$" ese es un comando solo para Linux y no lo reconocerá.

Abren una terminal (se van en la barra de herramientas a "Terminal" y luego "New terminal" pueden hacerlo en Power shell o Command prompt no hay mucha diferencia y ahí cambian el directorio y luego ejecutan el código

![image](https://user-images.githubusercontent.com/7990740/153771901-291f63b8-6b0d-4145-93e1-122f7f892ed3.png)

Así que cuando hagan el ejercicio de "Generación de excepciones" sepan que lo tienen que ejecutar así en notebook para que les de resultado.

![image](https://user-images.githubusercontent.com/7990740/153771948-629588e2-49cd-4ce3-a3fb-869d0df44c36.png)

Dejen mas tips o sugerencias aquí para quien lo necesite lo pueda ver y no olviden que mañana cuando suban sus katas al forms poner su repositorio en publico para que lo puedan ver, si no saben como les dejo el link donde lo explique 
[https://github.com/LaunchX-InnovaccionVirtual/CursoIntroPython/discussions/170#discussion-3871873](url)

**Comment by JavGom201** [comment]
Me ayudaron mucho estos tips, muchas gracias 👍

**Comment by bacopulos** [comment]
disculpen, yo no entiendo esta parte de la kata 10:
"A continuación, quita,ps el archivo config.txt y creamos un directorio denominado config.txt. Intentaremos llamar al archivo config.py para ver un error nuevo que debería ser similar al siguiente:"

si alguien pudiera ayudarme se lo agradeceria mucho

**Reply by VictorHugoChavez** [reply]
igual estoy en la misma duda

**Reply by DiegoPenalozaJuarez** [reply]
se debe crear un archivo llamado config.txt y ejecutas el codigo, posterior en la terminal ya sea cmd o en la terminal de Visual Code llamas al archivo config.py con el comando:
`py config.py` 
o tambien te puede servir:
`python3 config.py`

Y enseguida te debe salir el error que indica en la kata. Esta mal traducida esa parte y tiene errores de ortografia pero basicamente es eso.

**Reply by DavidMerino25** [reply]
Primero creas el archivo config.txt  pero después la instrucción dice que lo conviertas en un directorio es decir una carpeta, ejecutas config.py y listo, te mostrará el erro que marca la instrucción

**Comment by JellyRamirez** [comment]
Hola, ¿se sube algo de esta kata?

**Reply by DavidMerino25** [reply]
Todos los ejercicios de ese módulo es la kata

**Reply by JellyRamirez** [reply]
¡Gracias!
